src/sync/pull.rs

```diff
@@ -991,19 +1080,26 @@ mod tests {
                 );
                 // In a nop sync we except Beginpull to succeed but sync_head will
                 // be empty.
-                if c.exp_err.is_none() {
-                    assert!(&got_resp.as_ref().unwrap().sync_head.is_empty());
+                if c.exp_begin_try_pull_result.is_ok() {
+                    assert!(&result.as_ref().unwrap().sync_head.is_empty());
                 }
             }
 
             // Check that BeginTryPullResponse is filled like we would expect.
-            if c.exp_err.is_none() {
-                if let Ok(pull_response) = &c.pull_result {
-                    assert_eq!(
-                        pull_response.http_request_info,
-                        got_resp.unwrap().http_request_info
-                    );
-                }
+            assert_eq!(result.is_ok(), c.exp_begin_try_pull_result.is_ok());
+            if let Ok(result) = result {
+                // Do not check syncHead... it is not stable?
```

Comment:
Synchead is checked way up above starting around line 992. The hash is not stable so up there we check all its fields. Sorry the comment above was misleading!

Reply:
Yup. This comment is dead "code". I had it before I realized it was checked earlier.

I will update comments.